language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/AbstractCompletionProvider.java

```diff
@@ -607,4 +613,60 @@ private CompletionItem getModuleCompletionItem(String label, String insertText,
 
         return completionItems;
     }
+
+    protected List<LSCompletionItem> getNamedArgCompletionItems(BallerinaCompletionContext context,
+                                                            List<ParameterSymbol> params,
+                                                            SeparatedNodeList<FunctionArgumentNode> argumentNodeList) {
+        int cursorPosition = context.getCursorPositionInTree();
+        List<String> existingNamedArgs = new ArrayList<>();
+        int index = 1;
+        for (Node child : argumentNodeList) {
+            TextRange textRange = child.textRange();
+            int startOffset = textRange.startOffset();
+            int endOffset = textRange.endOffset();
+            if (startOffset > cursorPosition
+                    && child.kind() == SyntaxKind.POSITIONAL_ARG || child.kind() == SyntaxKind.REST_ARG) {
+                return Collections.emptyList();
+            } else if ((startOffset > cursorPosition || endOffset < cursorPosition)) {
+                if (child.kind() == SyntaxKind.NAMED_ARG) {
+                    existingNamedArgs.add(((NamedArgumentNode) child).argumentName().name().text());
+                } else if (child.kind() == SyntaxKind.POSITIONAL_ARG && index - 1 < params.size()) {
+                    ParameterSymbol parameterSymbol = params.get(index - 1);
+                    existingNamedArgs.add(parameterSymbol.getName().orElse(""));
+                }
```

Comment:
Shall we filter the existing fields separately? Otherwise the logic is complex

Reply:
You are suggesting moving this to separate method? will do